File: test/test_type/2.fun_pro/test0.pas
program test(input,output);
var sum:integer;
a:real;
diff:real;

function add(x, y: integer): real;
var result: real;
begin
  result := x + y;
end;

function minus(x, y: real): integer;
var result: integer;
begin
  result := x - y;
end;

begin
a := 3.5;
sum := add(a, 4); // 类型不匹配错误，传递实数参数给接受整数参数的函数
diff:=minus(a, 4.5);
end.

File: src/test/test_type/2.fun_pro/c_code/test0.c
#include <stdio.h>

float diff;
float a;
int sum;

float add(int x, int y)
{
	float _add;
	float result;
	result = x+y;
	return _add;
}

int minus(float x, float y)
{
	int _minus;
	int result;
	result = x-y;
	return _minus;
}

int main()
{
	a = 3.5;
	sum = add(a, 4);
	diff = minus(a, 4.5);
	return 0;
}